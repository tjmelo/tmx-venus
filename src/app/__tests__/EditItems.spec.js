import toEditItems, { toEditingItems } from '../modules/EditItems';

jest.mock('../modules/EditItems');

const params = {
  parentNodeReference: 'parentNodeReference',
  itemNodeReference: 'itemNodeReference',
  toEditedItemI: 'toEditedItemI',
  el: 'el',
};

describe('Should edit the value of an item', () => {
  it('Should edit an item', () => {
    const testToEditItems = jest.fn();

    const testImplementationMock = () => toEditItems();

    testToEditItems.mockImplementation(testImplementationMock);

    // do
    testToEditItems();

    // then
    expect(testToEditItems).toHaveReturned();
    expect(testToEditItems).toHaveBeenCalledTimes(1);

    // Expects fidelity with scope of function
    expect(testToEditItems).toMatchSnapshot();
  });

  it('Should edit as a status', () => {
    const testEditingItemsStatus = jest.fn();

    testEditingItemsStatus.mockImplementation((param, status) => toEditingItems(param, status));

    // do
    testEditingItemsStatus(params, true);

    // then
    expect(testEditingItemsStatus).toHaveReturned();
    expect(testEditingItemsStatus).toHaveBeenCalledTimes(1);

    // Expects fidelity with scope of function
    expect(testEditingItemsStatus).toMatchSnapshot();
  });
});
