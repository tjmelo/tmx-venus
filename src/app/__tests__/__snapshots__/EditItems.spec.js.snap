// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should edit the value of an item Should edit an item 1`] = `
[MockFunction] {
  "calls": [
    [],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Should edit the value of an item Should edit as a status 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "el": "el",
        "itemNodeReference": "itemNodeReference",
        "parentNodeReference": "parentNodeReference",
        "toEditedItemI": "toEditedItemI",
      },
      true,
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
